client/components/domains/use-your-domain-step/index.jsx

```diff
@@ -1,469 +0,0 @@
-import { isPlan } from '@automattic/calypso-products';
-import page from '@automattic/calypso-router';
-import { Card, Button, Gridicon } from '@automattic/components';
-import { localizeUrl } from '@automattic/i18n-utils';
-import { withShoppingCart } from '@automattic/shopping-cart';
-import {
-	CALYPSO_HELP_WITH_HELP_CENTER,
-	INCOMING_DOMAIN_TRANSFER,
-	MAP_EXISTING_DOMAIN,
-} from '@automattic/urls';
-import { formatCurrency, localize } from 'i18n-calypso';
-import { get, isEmpty } from 'lodash';
-import PropTypes from 'prop-types';
-import { stringify } from 'qs';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import migratingHostImage from 'calypso/assets/images/illustrations/migrating-host-diy.svg';
-import themesImage from 'calypso/assets/images/illustrations/themes.svg';
-import QueryProducts from 'calypso/components/data/query-products-list';
-import HeaderCake from 'calypso/components/header-cake';
-import {
-	isDomainBundledWithPlan,
-	isDomainMappingFree,
-	isNextDomainFree,
-} from 'calypso/lib/cart-values/cart-items';
-import {
-	getDomainPrice,
-	getDomainProductSlug,
-	getDomainTransferSalePrice,
-} from 'calypso/lib/domains';
-import withCartKey from 'calypso/my-sites/checkout/with-cart-key';
-import { recordTracksEvent } from 'calypso/state/analytics/actions';
-import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
-import {
-	DOMAINS_WITH_PLANS_ONLY,
-	NON_PRIMARY_DOMAINS_TO_FREE_USERS,
-} from 'calypso/state/current-user/constants';
-import { currentUserHasFlag, getCurrentUser } from 'calypso/state/current-user/selectors';
-import { errorNotice } from 'calypso/state/notices/actions';
-import { getProductsList } from 'calypso/state/products-list/selectors';
-import { getSelectedSite } from 'calypso/state/ui/selectors';
-
-import './style.scss';
-
-const noop = () => {};
-
-class UseYourDomainStep extends Component {
-	static propTypes = {
-		analyticsSection: PropTypes.string.isRequired,
-		basePath: PropTypes.string,
-		cart: PropTypes.object,
-		domainsWithPlansOnly: PropTypes.bool,
-		primaryWithPlansOnly: PropTypes.bool,
-		goBack: PropTypes.func,
-		initialQuery: PropTypes.string,
-		isSignupStep: PropTypes.bool,
-		mapDomainUrl: PropTypes.string,
-		transferDomainUrl: PropTypes.string,
-		onRegisterDomain: PropTypes.func,
-		onTransferDomain: PropTypes.func,
-		onSave: PropTypes.func,
-		selectedSite: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
-		forcePrecheck: PropTypes.bool,
-	};
```

Comment:
I was able to remove a lot of props compared to the previous implementation:
- some of them came from `connect`ing the component to the Redux Global state, and were replaced by hooks;
- some of them are actually not currently used, and I guess were not cleaned up over time

Reply:
In the Git history I see big refactorings (#25573 is an example) where large chunks of code were moved around between `use-your-domain-step`, `register-domain-step`, `transfer-domain-step` etc. That explains how many props and state fields got left behind unused.